core/src/main/java/ai/timefold/solver/core/api/score/buildin/hardmediumsoftbigdecimal/HardMediumSoftBigDecimalScore.java

```diff
@@ -53,15 +55,18 @@ public static HardMediumSoftBigDecimalScore parseScore(String scoreString) {
         return ofUninitialized(initScore, hardScore, mediumScore, softScore);
     }
 
-    public static HardMediumSoftBigDecimalScore ofUninitialized(int initScore, BigDecimal hardScore, BigDecimal mediumScore,
+    public static @NonNull HardMediumSoftBigDecimalScore ofUninitialized(int initScore, BigDecimal hardScore,
+            BigDecimal mediumScore,
             BigDecimal softScore) {
         if (initScore == 0) {
             return of(hardScore, mediumScore, softScore);
         }
         return new HardMediumSoftBigDecimalScore(initScore, hardScore, mediumScore, softScore);
     }
 
-    public static HardMediumSoftBigDecimalScore of(BigDecimal hardScore, BigDecimal mediumScore, BigDecimal softScore) {
+    // TODO: params nonnull?
```

Comment:
IMO yes. Let's make sure we propagate this to the other `BigDecimal` score types as well.